File: examples/nodejs/kafka-http/kafka.js
var request = require('request')

exports.pub = function(event, callback) {
  var topic = process.env.KAFKA_TOPIC
  var host = process.env.KAFKA_HOST
  var user = process.env.KAFKA_USER
  var password = process.env.KAFKA_PASSWORD

  if (!event.data) return callback('data not found')

  var uri = host+'/topic/'+topic+'?format=raw'
  var req = {
    method: 'POST',
    auth: {
      user: user,
      password: password
    },
    json: event.data,
  }

  request.post(uri, req, function (error, resp) {
      if (error) return callback(error)
      callback(null, JSON.stringify(resp.body))
    })
}

File: examples/go/kafka-http/kafka.go
package main

import (
	"bytes"
	"errors"
	"io/ioutil"
	"net/http"
	"os"

	"github.com/ovhlabs/functions/go-sdk/event"
)

func Pub(event event.Event) (string, error) {
	topic := os.Getenv("KAFKA_TOPIC")
	host := os.Getenv("KAFKA_HOST")
	user := os.Getenv("KAFKA_USER")
	password := os.Getenv("KAFKA_PASSWORD")

	if event.Data == "" {
		return "", errors.New("data not found")
	}

	// prepares the request
	client := &http.Client{}
	url := host + "/topic/" + topic + "?format=raw"
	req, err := http.NewRequest("POST", url, bytes.NewBuffer([]byte(event.Data)))
	if err != nil {
		return "", err
	}
	req.SetBasicAuth(user, password)

	// sends the request
	resp, err := client.Do(req)
	if err != nil {
		return "", err
	}
	defer resp.Body.Close()

	body, err := ioutil.ReadAll(resp.Body)

	return string(body), err
}
